### PR TITLE
[AIRFLOW-1840] Make celery configuration congruent with Celery 4

### PR DESCRIPTION
**Pull request by Fokko**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

This PR consists off:
- Rename `CELERY_RESULT_BACKEND` to `RESULT_BACKEND`. This aligns with the naming of Celery itself: http://docs.celeryproject.org/en/latest/userguide/configuration.html#new-lowercase-settings. This makes the configuration more transparant.
- Make sure that the celery_ssl_active is set in the config, otherwise you get a lot of errors in the logs:
```
[2017-11-22 15:10:01,779] {configuration.py:206} WARNING - section/key [celery/celery_ssl_active] not found in config
[2017-11-22 15:10:01,780] {default_celery.py:40} WARNING - Celery Executor will run without SSL
```

I've tested the configuration on acceptance using Celery 4.1

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-1840] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1840

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by villasv** [comment]
Depending on how fast dependency updates are merged, I'd rather wait for 4.1.1 because 4.1.0 doesn't install the correct `boto` version with `celery[sqs]`. But I'm not sure how long that's going to take, and if going from 4.1.0 to 4.1.1 can be fast, it doesn't really matter.

**Comment by Fokko** [comment]
@villasv Thanks for the feedback, lets wait for 4.1.1 or 4.2.0 then.

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -303,6 +308,12 @@ default_queue = default
 # Import path for celery configuration options
 celery_config_options = airflow.config_templates.default_celery.DEFAULT_CELERY_CONFIG
 
+# In case of using SSL
+CELERY_SSL_ACTIVE = False
```
Comment (bolkedebruin) [inline airflow/config_templates/default_airflow.cfg]: Why keep these uppercase?

**Comment by bolkedebruin** [comment]
Can you rename your commit to "Make celery configuration congruent with celery 4" aside from the small nit that lgtm

**Comment by bolkedebruin** [comment]
This is backward incompatible please provide upgrade instructions or backward compatibility

**Comment by Fokko** [comment]
I've updated UPDATING.md with the changed configuration parameters.

File: airflow/config_templates/default_airflow.cfg [hunk 2]
```diff
@@ -317,6 +323,12 @@ celery_config_options = airflow.config_templates.default_celery.DEFAULT_CELERY_C
 # ETA you’re planning to use. Especially important in case of using Redis or SQS
 visibility_timeout = 21600
 
+# In case of using SSL
```
Comment (bolkedebruin) [inline airflow/config_templates/default_airflow.cfg]: Why keep this in uppercase? Why keep it prepended with `celery_`?

Comment (Fokko) [reply airflow/config_templates/default_airflow.cfg]: Good point Bolke, I've removed the `celery_` prefixes.

Comment (bolkedebruin) [reply airflow/config_templates/default_airflow.cfg]: Still uppercase? Please use lowercase as we do everwhere else and celery4 moved to lowercase.

Comment (Fokko) [reply airflow/config_templates/default_airflow.cfg]: Good point @bolkedebruin, I've lowercased the keys

**Comment by bolkedebruin** [comment]
LGTM

**Comment by Fokko** [comment]
Thanks for the feedback

**Comment by jgao54** [comment]
@Fokko , while I was re-boostrapping a new airflow instance in VM locally, I noticed that there are a few places with `celeryd_concurrency` that still needs to be converted to `worker_concurrency` in the codebase, specifically:
- airflow/bin/cli.py
- airflow/config_templates/default_test.cfg
- airflow-source/docs/configuration.rst
- airflow-source/scripts/ci/airflow_travis.cfg

**Comment by Fokko** [comment]
Thanks for spotting @jgao54 

I've created a new PR https://github.com/apache/incubator-airflow/pull/2870

**Comment by ashb** [comment]
This makes it a non-backwards compatible change going from 1.9.0 to 1.9.1. It would be nice if we supported both styles for at least one point release.

Django's approach of "if you have no deprecation warnings now then if you upgrade it will still work" is one I like. It makes upgrading easier to my mind if you can upgrade then fix the issues, rather than having to upgrade the deployment and fix issues in lock-step.

API wise it could look like `conf.get('celery', 'worker_concurrency', deprecated='celeryd_concurrency')`

**Comment by bolkedebruin** [comment]
I’m not sure if I would consider this to be really backward incompatible. It is not so that you have to change a 100 dags. 

Your suggestion for the configuration does sound like a good idea.

**Comment by ashb** [comment]
My idea around "backwards compatible" is I can take a currently running system that isn't issuing  any deprecation warnings, upgrade the version of airflow and have it run without making any other changes. I realise this is quite an extreme view though.

This is better discussed on the mailing list. I'll do that (though next week).

**Comment by bolkedebruin** [comment]
Gotcha. My idea of backwards compatible is more along the lines of FreeBSD. API/ABI compatible, but config changes can happen and are in UPDATING.

**Comment by cloneluke** [comment]
@tedmiston - when do you plan on removing this pin?
celery 4.2 is in RC2
https://github.com/celery/celery/releases  @bolkedebruin  @Fokko

**Comment by tedmiston** [comment]
Hey @cloneluke - I replied back over on https://github.com/astronomerio/incubator-airflow/pull/23.

**Comment by Fokko** [comment]
@cloneluke I'm following the discussions on the Celery release cycle, but the release process it is moving very slowly 😞

**Comment by Fokko** [comment]
@tedmiston feel free to backport the PR onto Apache Airflow master :-)

**Comment by cloneluke** [comment]
I have tested with celery 4.1.1 and airflow seems to be fine, 4.2 still a release candidate.

**Comment by tedmiston** [comment]
It looks like 4.2 RC or not should "just work" with apache/incubator-airflow unless I'm missing something.

https://github.com/apache/incubator-airflow/blob/7c34354427f8c047b2cfb3f472a38bc50fe91d45/setup.py#L118-L121

(We had pinned our fork to 4.0.2.)

I'm happy to copy over this commit https://github.com/astronomerio/incubator-airflow/commit/4fb708cb3ea4284bc66ea992b3b0fe575240bdaa which guards against the next major release of Celery like so `'celery>=4.0.2, <5'` as some other dependencies already do in setup.py.

@Fokko Would it be useful for others for me to send a PR back with that change?

**Comment by cloneluke** [comment]
@tedmiston - Agreed - We pinned our cookbook like this:
'>=4.1.0, <4.2.0'
https://github.com/bahchis/airflow-cookbook/pull/22/files#diff-63da1eb5d35a0fe1016e58ed5c24ad9eR20

**Comment by Fokko** [comment]
@tedmiston @cloneluke As far as I understand, Celery 4.1.0 is broken. So my suggestion would be `>4.1.0, <4.2.0`. I would welcome this PR.

**Comment by tedmiston** [comment]
I just sent over a Jira issue + PR in https://issues.apache.org/jira/browse/AIRFLOW-2544 / https://github.com/apache/incubator-airflow/pull/3439

**Comment by Fokko** [comment]
Thanks

**Pull request closed without merging**